# Big data sets in python
(Well medium really)

### A few imports

In [3]:
# Pandas for data handling, but let's call it pd for short
import pandas as pd
# Numpy also for data handling, let's call it np
import numpy as np
# toyplot for ... plotting
import toyplot as tp

### Get the data

In [4]:
# Get the height data
data = pd.read_csv("Heights.csv")
# Look at the first five values
data.head(5)

height
0     180
1     166
2     174
3     181
4     177

### How much is there?

In [10]:
len(data)

1500

## Task 1

* Our `data` object has methods `mean()` and `median()`. Use them to comment on the distribution of heights.

We can use `numpy`'s `histogram` function to group the data

In [11]:
np.histogram(data)

(array([ 12,  45, 107, 275, 319, 297, 286, 120,  29,  10]),
 array([ 157. ,  161.3,  165.6,  169.9,  174.2,  178.5,  182.8,  187.1,
         191.4,  195.7,  200. ]))

And `toyplot`'s `bars` to visualise the distribution

In [12]:
tp.bars(np.histogram(data))

(<toyplot.canvas.Canvas at 0x7f04aaf115f8>,
 <toyplot.mark.BarMagnitudes at 0x7f04aa618b70>)

160 170 180 190 200 0 100 200 300 
 
 
 Save as .csv

We can change the number of bars with the `bins` argument

In [13]:
tp.bars(np.histogram(data,bins=3))

(<toyplot.canvas.Canvas at 0x7f04e42fd6a0>,
 <toyplot.mark.BarMagnitudes at 0x7f04aa618dd8>)

160 170 180 190 200 0 500 1000 
 
 
 Save as .csv

## Task 2

* Experiment with different values of `bins` above. What happens to the histogram? What sort of shape is it?
* Does this confirm what you expected from looking at the mean and median earlier?

In [7]:
100*len(data[data.height>195])/len(data)

0.6666666666666666

## Task 3

The syntax above selects all the heights in the table which are above $195$ cm.

* How many are there? What percentage of the data does this represent
* What percentage of the heights are below $170$ cm?
* What percentage of the heights are greater than your height?
* What percentage of the heights are shorter than the median? The mean?
* Find a height which is taller than $75$% of the data

In [17]:
 len(data[data.height<=185])/len(data)

0.8326666666666667

## Task 4

* Find a height which is taller than $25$% of the data
* Find the lower quartile

`data.quantile(0.25)=`

In [18]:
data.quantile(0.75)

height    183.0
Name: 0.75, dtype: float64

Find
* the height which is taller than $95$% of the data (the 95th percentile)
* the height which is taller than $\dfrac{2}{5}$ of the data (the second quintile)
* the lowest decile and the highest decile

In [19]:
data.quantile(0.95)

height    190.0
Name: 0.95, dtype: float64